pkg/apis/agent/v1alpha1/validations.go

```diff
@@ -137,6 +143,27 @@ func checkNoDowngrade(prev, curr *Agent) field.ErrorList {
 	return commonv1.CheckNoDowngrade(prev.Spec.Version, curr.Spec.Version)
 }
 
+// checkPVCchanges ensures no PVCs are changed, as volume claim templates are immutable in StatefulSets.
+func checkPVCchanges(current, proposed *Agent) field.ErrorList {
+	var errs field.ErrorList
+	if current == nil || proposed == nil {
+		return errs
+	}
+
+	// need to check if current and proposed are both statefulsets
+	if current.Spec.StatefulSet == nil || proposed.Spec.StatefulSet == nil {
+		return errs
+	}
+
+	// checking semantic equality here allows providing PVC storage size with different units (eg. 1Ti vs. 1024Gi).
+	if !apiequality.Semantic.DeepEqual(current.Spec.StatefulSet.VolumeClaimTemplates, proposed.Spec.StatefulSet.VolumeClaimTemplates) {
+		errs = append(errs, field.Invalid(field.NewPath("spec").Child("volumeClaimTemplates"),
```

Comment:
nit:
```suggestion
		errs = append(errs, field.Invalid(field.NewPath("spec").Child("statefulSet.").Child("volumeClaimTemplates"),
```

Reply:
oh my 😄 I adore the fact that I paid attention to the golang struct spec.Statefulset but not in the field path.. thanks @barkbay